packages/client/ui/react-ui/src/providers/CrossmintWalletProvider.tsx

```diff
@@ -54,31 +71,69 @@ export function CrossmintWalletProvider({
             return { startedCreation: false, reason: `Jwt not set in "CrossmintProvider".` };
         }
 
-        const internalCall = async () => {
+        if (enablePasskeyPrompt && config.signer && "type" in config.signer && config.signer.type === "PASSKEY") {
+            if (config.onCreateWalletPasskeyCallback == null) {
+                config.onCreateWalletPasskeyCallback = () =>
+                    new Promise<void>((resolve) => {
+                        setPasskeyPromptState({
+                            type: "create-wallet",
+                            open: true,
+                            primaryActionOnClick: () => {
+                                setPasskeyPromptState({ open: false });
+                                resolve();
+                            },
+                        });
+                    });
+            }
+
+            if (config.onErrorCreateWalletCallback == null) {
+                config.onErrorCreateWalletCallback = () => {
+                    return new Promise<void>((resolve) => {
+                        setPasskeyPromptState({
+                            type: "create-wallet-error",
+                            open: true,
+                            primaryActionOnClick: async () => {
+                                setPasskeyPromptState({ open: false });
+                                // When retrying, we don't want to show the passkey prompt again so unset the wallet creation callback
+                                config.onCreateWalletPasskeyCallback = undefined;
```

Comment:
How does this line work?

Reply:
Wouldn't this only change this for a single render?

Reply:
The config modification is local to the current execution of getOrCreateWallet due to Javascripts  lexical scoping. It doesn't affect the global state or future calls to the function. Therefore, the call right after will maintain the state due to the lexical scope and the onCreateWalletPasskeyCallback fn will still be undefined.

Reply:
removed this